File: windows/DMShot.Tests/UpdateSnoozeStoreTests.cs
using System.IO;
using DMShot.Settings;
using DMShot.Update;
using Xunit;

namespace DMShot.Tests;

/// A "Later" on the update prompt has to outlive the process — otherwise every
/// relaunch would ask again and the snooze would be decorative.
/// mac parity: UpdateSnoozeStoreTests.swift.
public class UpdateSnoozeStoreTests : IDisposable
{
    private readonly string _path = Path.Combine(Path.GetTempPath(), $"dmshot-snooze-{Guid.NewGuid():N}.json");
    public void Dispose() { if (File.Exists(_path)) File.Delete(_path); }

    [Fact]
    public void DefaultsToNoSnooze()
    {
        var s = new Settings.Settings();
        Assert.Null(UpdatePrompt.SnoozeFrom(s.UpdateSnoozeVersion, s.UpdateSnoozeUntil));
    }

    [Fact]
    public void SurvivesARestart()
    {
        var until = new DateTimeOffset(2026, 8, 3, 12, 0, 0, TimeSpan.Zero);
        var store = new SettingsStore(_path);
        var settings = store.Load();
        settings.UpdateSnoozeVersion = "1.2.3";
        settings.UpdateSnoozeUntil = until;
        store.Save(settings);

        var reloaded = new SettingsStore(_path).Load();
        var snooze = UpdatePrompt.SnoozeFrom(reloaded.UpdateSnoozeVersion, reloaded.UpdateSnoozeUntil);
        Assert.NotNull(snooze);
        Assert.Equal("1.2.3", snooze!.Value.Version);
        Assert.Equal(until, snooze.Value.Until);
    }

    [Fact]
    public void ClearingRemovesTheSnooze()
    {
        var store = new SettingsStore(_path);
        var settings = store.Load();
        settings.UpdateSnoozeVersion = "1.2.3";
        settings.UpdateSnoozeUntil = DateTimeOffset.UtcNow;
        store.Save(settings);

        settings.UpdateSnoozeVersion = "";
        settings.UpdateSnoozeUntil = null;
        store.Save(settings);

        var reloaded = new SettingsStore(_path).Load();
        Assert.Null(UpdatePrompt.SnoozeFrom(reloaded.UpdateSnoozeVersion, reloaded.UpdateSnoozeUntil));
    }
}

File: mac/Tests/DMShotTests/UpdateSnoozeStoreTests.swift
import XCTest
@testable import DMShot

/// A "Later" on the update prompt has to outlive the process — otherwise every
/// relaunch would ask again and the snooze would be decorative.
final class UpdateSnoozeStoreTests: XCTestCase {
    private var suite: String!
    private var defaults: UserDefaults!

    override func setUp() {
        super.setUp()
        suite = "dmshot.tests.snooze.\(UUID().uuidString)"
        defaults = UserDefaults(suiteName: suite)!
    }

    override func tearDown() {
        defaults.removePersistentDomain(forName: suite)
        super.tearDown()
    }

    func testDefaultsToNoSnooze() {
        XCTAssertNil(AppSettingsStore(defaults: defaults).updateSnooze)
    }

    func testSurvivesARestart() {
        let until = Date(timeIntervalSince1970: 1_700_000_000)
        AppSettingsStore(defaults: defaults).updateSnooze = UpdateSnooze(version: "1.2.3", until: until)

        let reopened = AppSettingsStore(defaults: defaults).updateSnooze
        XCTAssertEqual(reopened?.version, "1.2.3")
        XCTAssertEqual(reopened?.until, until)
    }

    func testClearing() {
        let store = AppSettingsStore(defaults: defaults)
        store.updateSnooze = UpdateSnooze(version: "1.2.3", until: Date())
        store.updateSnooze = nil
        XCTAssertNil(AppSettingsStore(defaults: defaults).updateSnooze)
    }

    func testHalfWrittenSnoozeIsIgnored() {
        defaults.set("1.2.3", forKey: AppSettingsStore.updateSnoozeVersionKey)
        XCTAssertNil(AppSettingsStore(defaults: defaults).updateSnooze)
    }
}
